automation_tools/satellite6/upgrade/tests/existence.py

```diff
@@ -0,0 +1,185 @@
+"""Helper functions and variables to test entity existence and associations
+post upgrade
+"""
+
+import os
+import pickle
+from automation_tools.satellite6.upgrade.tasks import csv_reader
+
+# Components for which the post upgrade existence will be validated,
+# non_orged - The components where org is not required to get the data about
+# orged - The components where org is required to get the data about
+components = {
+    'non_orged':
+    [
+        'organization',
+        'host',
+        'capsule',
+        'role',
+        'compute-resource',
+        'user',
+        'subnet',
+        'settings',
+        'os',
+        'hostgroup',
+        'domain'
+    ],
+    'orged':
+    [
+        'content-view',
+        'activation-key'
+    ]
+}
+
+
+def set_datastore(datastore):
+    """
+    Creates a pickle file with all the satellite components data
```

Comment:
why use pickle? I think JSON format is better here because 1) it is human readable 2) you can craft it manually if needed 3) easy to parse and generate 4) also in Python's std lib 5) the data is only dicts, lists and strings